lib/internal/util/iterable-weak-map.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const {
+  SafeSet,
+  SafeWeakMap,
+  Symbol,
+} = primordials;
+
+// This class is modified from the example code in the WeakRefs specification:
+// https://github.com/tc39/proposal-weakrefs
+// Licensed under ECMA's MIT-style license, see:
+// https://github.com/tc39/ecma262/blob/master/LICENSE.md
+class IterableWeakMap {
+  #weakMap = new SafeWeakMap();
+  #refSet = new SafeSet();
+  // TODO(aduh95): Add FinalizationRegistry to primordials
```

Comment:
to call it out explicitly: i think these primordials todos should be considered blockers

Reply:
@ljharb @aduh95, how about the approach of exposing the `makeSafe` from primordials, and freezing `WeakRef` and `FinalizationRegistry` when `iterable-weak-map.js` is loaded:

```js
// TODO(aduh95): Add WeakRef to primordials
const SafeWeakRef = makeSafe(
  globalThis.WeakRef,
  class SafeWeakRef extends globalThis.WeakRef {}
);

// This class is modified from the example code in the WeakRefs specification:
// https://github.com/tc39/proposal-weakrefs
// Licensed under ECMA's MIT-style license, see:
// https://github.com/tc39/ecma262/blob/master/LICENSE.md
class IterableWeakMap {
// ...
}}
```

`source_map_cache.js` pretty early on in the application lifecycle, _I think, before any userland code.

Reply:
That seems fine, but I’m not sure what’s wrong with eagerly freezing it, like all the other primordials?

Reply:
When the primordials are prepared, we only have access to a subset of builtins. V8 does not provide us the ones that can be disabled with flags by the user.

Reply:
ahhhh thanks, that makes sense.